import "vehicle_status.proto";
import "position_status.proto";
import "soc_status.proto";

option java_package = "com.nextev.cvs_proto.protobuf.events";
option java_outer_classname="ConnectionStatusMessage";

message ConnectionStatusMsg {
	optional string vid = 1; //vehicle id
	optional uint64 sample_ts = 2;
	optional connectionStatus connection_status = 3;
	optional string process_id = 4;
	optional PositionStatus position_status = 5;
	optional VehicleStatus vehicle_status = 6;
	optional SOCStatus soc_status = 7;

	message connectionStatus {
		optional EcuType ecu_type = 1;
		optional Status status = 2;
		optional string latest_msg_type = 3;
	}

	enum EcuType {
		CGW = 0;
		CDC = 1;
		ADC = 2;
	}

	enum Status {
		OFFLINE = 0;
		ONLINE = 1;
		CONNECTION_LOST = 2;
	}
}